src/log/log_private.h

```diff
@@ -233,7 +233,7 @@ struct __wti_log {
     int32_t pool_index;                             /* Index into slot pool */
     size_t slot_buf_size;                           /* Buffer size for slots */
 #ifdef HAVE_DIAGNOSTIC
-    uint64_t write_calls; /* Calls to log_write */
+    wt_shared uint64_t write_calls; /* Calls to log_write */
```

Comment:
Quick fix to resolve a TSan error firing in the examples-c-tsan variant.

I've stress tests `examples-c-tsan` 100 times and it didn't hit any errors (with this change and disabling `ex_backup`) so I don't think there's any risk in adding it to PR tests.